### PR TITLE
Logictest projections

### PR DESCRIPTION
**Pull request by thorfour**
This adds Projection support to Logictests as well as adding support for additional schemas.

### CODE REVIEW

File: logictest/schemas/default.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+    "name": "default",
```
Comment (asubiotto) [inline logictest/schemas/default.json:2]: Love this idea. Is there a reason to add an additional json parsing layer? In my mind we'd do something like:
```
var schemas = map[string]*dynparquet.Schema{
    "default": dynparquet.NewSampleSchema(),
    "bools": dynparquet.SchemaFromDefinition(
        < your bools def here >
    )
}
```

I'm open to this json idea, just curious.

File: sqlparse/visitor.go [hunk 2]
```diff
@@ -62,6 +62,8 @@ func (v *astVisitor) leaveImpl(n ast.Node) error {
 				v.exprStack[0],
 				v.exprStack[1:]...,
 			)
+		default:
```
Comment (asubiotto) [inline sqlparse/visitor.go:65]: Is it correct to have this as the `default` case? I guess previously we just returned all columns, so we do want to do a Project in all cases, is that right?

File: logictest/runner.go [hunk 3]
```diff
@@ -105,10 +105,10 @@ func (r *Runner) handleCreateTable(ctx context.Context, c *datadriven.TestData)
 	var schema *dynparquet.Schema
 	for _, arg := range c.CmdArgs {
 		if arg.Key == "schema" {
-			if len(arg.Vals) != 1 && arg.Vals[0] != "default" {
+			if len(arg.Vals) != 1 {
 				return "", fmt.Errorf("createtable: unexpected schema values %v", arg.Vals)
 			}
-			schema = r.defaultSchema
+			schema = r.schemas[arg.Vals[0]]
```
Comment (asubiotto) [inline logictest/runner.go:111]: nit: return an error if schema is not found

**Review by asubiotto — COMMENTED** [review]
nice!

File: logictest/schemas/default.json (hunk 1 above)
Comment (thorfour) [reply logictest/schemas/default.json:2]: I typically don't put configuration settings into code, but I don't have strong feelings against doing it that way

File: logictest/runner.go (hunk 3 above)
Comment (thorfour) [reply logictest/runner.go:111]: We do below where we check if schema is `nil`

File: sqlparse/visitor.go (hunk 2 above)
Comment (thorfour) [reply sqlparse/visitor.go:65]: I think we'd want to extend this to support something like `select *` in the future, but I think for what we have now this is okay?

File: logictest/logic_test.go [hunk 4]
```diff
@@ -2,6 +2,10 @@ package logictest
 
 import (
 	"context"
+	"encoding/json"
+	"io/fs"
+	"io/ioutil"
```
Comment (metalmatze) [inline logictest/logic_test.go:7]: Deprecated. Use the newer APIs.

Comment (thorfour) [reply logictest/logic_test.go:7]: thanks; old habits die hard.

**Review by brancz — APPROVED** [review]

**Pull request merged**